pkg/controller/gitsourceanalysis/gitsourceanalysis_controller_test.go

```diff
@@ -0,0 +1,291 @@
+package gitsourceanalysis
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"testing"
+
+	"github.com/redhat-developer/devconsole-api/pkg/apis/devconsole/v1alpha1"
+	"github.com/redhat-developer/devconsole-git/pkg/git/detector/build"
+	"github.com/redhat-developer/devconsole-git/pkg/git/repository"
+	"github.com/redhat-developer/devconsole-git/pkg/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/h2non/gock.v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	pathToTestDir  = "../../test"
```

Comment:
@MatousJobanek Should I get `test` files from `vendor` here ? What do you you think ?